<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <link rel="stylesheet" type= "text/css" href="../styles.css"> <!-- Include your CSS file here -->
    <title>Try...Catch </title>

</head>
<body>
    <div class="wrapper">
        <div id="stars"></div>
        <div id="stars2"></div>
        <div id="stars3"></div>
    </div>
    <div class="content">
        <h1 id="trycatch">Try...Catch</h1>
        <p>The life of a progammer is full of bugs and errors. Debugging is a beast of its own ([]../[]), but errors due to wrong input or other unforeseable things can be handled by throwing an <strong>exception</strong>.</p>
        <h2 id="java-try-and-catch">Java Try and Catch</h2>
        <p>The <code>try</code> statement allows you to define a block of code to be tested for errors while it is being executed.</p>
        <p>The <code>catch</code> statement allows you to define a block of code to be executed, if an error occurs in the try block.</p>
        <p>The <code>try</code> and <code>catch</code> keywords come in pairs:</p>
        <div class="sourceCode" id="cb1"><pre class="sourceCode java"><code class="sourceCode java"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true"></a><span class="kw">public</span> <span class="kw">class</span> Main {</span>
        <span id="cb1-2"><a href="#cb1-2" aria-hidden="true"></a>    <span class="kw">public</span> <span class="dt">static</span> <span class="dt">void</span> <span class="fu">main</span>(<span class="bu">String</span>[ ] args) {</span>
        <span id="cb1-3"><a href="#cb1-3" aria-hidden="true"></a>        <span class="kw">try</span> {</span>
        <span id="cb1-4"><a href="#cb1-4" aria-hidden="true"></a>            <span class="dt">int</span>[] myNumbers = {<span class="dv">1</span>, <span class="dv">2</span>, <span class="dv">3</span>};</span>
        <span id="cb1-5"><a href="#cb1-5" aria-hidden="true"></a>            <span class="bu">System</span>.<span class="fu">out</span>.<span class="fu">println</span>(myNumbers[<span class="dv">10</span>]);</span>
        <span id="cb1-6"><a href="#cb1-6" aria-hidden="true"></a>        } <span class="kw">catch</span> (<span class="bu">Exception</span> e) {</span>
        <span id="cb1-7"><a href="#cb1-7" aria-hidden="true"></a>            <span class="bu">System</span>.<span class="fu">out</span>.<span class="fu">println</span>(<span class="st">&quot;Something went wrong.&quot;</span>);</span>
        <span id="cb1-8"><a href="#cb1-8" aria-hidden="true"></a>        }</span>
        <span id="cb1-9"><a href="#cb1-9" aria-hidden="true"></a>    }</span>
        <span id="cb1-10"><a href="#cb1-10" aria-hidden="true"></a>}</span></code></pre></div>
        <p>In the above example, <code>myNumbers</code> is an array with a length of 3; therefore, trying to reference an index of 10 will generate an error: <code>ArrayIndexOutOfBoundsException: 10</code>. Thus, the use of a <code>try</code> and <code>catch</code> statement can be used to "rewrite" the error message. Not only that, but if you implement the <code>finally</code> statement after the <code>catch</code>, you can execute code after the <code>try...catch</code> regardless of the result:</p>
        <div class="sourceCode" id="cb2"><pre class="sourceCode java"><code class="sourceCode java"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true"></a><span class="kw">finally</span> {</span>
        <span id="cb2-2"><a href="#cb2-2" aria-hidden="true"></a>    <span class="bu">System</span>.<span class="fu">out</span>.<span class="fu">println</span>(<span class="st">&quot;The &#39;try catch&#39; is finished.&quot;</span>);</span>
        <span id="cb2-3"><a href="#cb2-3" aria-hidden="true"></a>}</span></code></pre></div>
        <h2 id="references">References</h2>
        <p><a href="https://www.w3schools.com/java/java_try_catch.asp">W3Schools Java Exceptions</a></p>
        <h2 id="tags">Tags</h2>
        <p>#java #software #programming</p>
    </div>
</body>
</html>
